### PR TITLE
[ui] Add a line edit to type character for the font marker

### PR DESCRIPTION
**Pull request by nirvn**
## Description
A GIF worth a thousand words:
peekFKEU0Z

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [ ] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and contain sufficient information in the commit message to be documented
- [ ] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] New unit tests have been added for core changes
- [ ] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTING.md#contributing-to-qgis) before each commit

### CODE REVIEW

File: src/gui/symbology/qgssymbollayerwidget.h [hunk 1]
```diff
@@ -932,8 +932,20 @@ class GUI_EXPORT QgsFontMarkerSymbolLayerWidget : public QgsSymbolLayerWidget, p
...
+     * \param chr the char
+     */
     void setCharacter( QChar chr );
 
+    /**
+     * Set the font marker character from a text string.
+     * \param text the text string
+     * \since QGIS 3.8
+     */
+    void setCharacterFromText( QString text );
```
Comment (m-kuhn) [inline src/gui/symbology/qgssymbollayerwidget.h]:
```suggestion
    void setCharacterFromText( const QString &text );
```

**Comment by m-kuhn** [comment]
Nice!
What do you think, could this just appear on top of the table when the user starts typing?

**Comment by nirvn** [comment]
@m-kuhn , -1 to hide the line edit by default. +1 to add support for typing a character when the char picker widget is focused.

**Comment by nirvn** [comment]
@m-kuhn , comments addressed, thanks.

**Pull request merged**